pkg/user/admin/is_admin.go

```diff
@@ -1,26 +1,138 @@
 package admin
 
 import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/internal/common"
 	"github.com/Layr-Labs/eigenlayer-cli/pkg/internal/common/flags"
 	"github.com/Layr-Labs/eigenlayer-cli/pkg/telemetry"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/utils"
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/elcontracts"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	eigenSdkUtils "github.com/Layr-Labs/eigensdk-go/utils"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+
 	"github.com/urfave/cli/v2"
 )
 
-func IsAdminCmd() *cli.Command {
-	isAdmin := &cli.Command{
+type IsAdminReader interface {
+	IsAdmin(
+		ctx context.Context,
+		accountAddress gethcommon.Address,
+		pendingAdminAddress gethcommon.Address,
+	) (bool, error)
+}
+
+func IsAdminCmd(readerGenerator func(logging.Logger, *isAdminConfig) (IsAdminReader, error)) *cli.Command {
+	cmd := &cli.Command{
 		Name:      "is-admin",
 		Usage:     "user admin is-admin --account-address <AccountAddress> --caller-address <CallerAddress>",
 		UsageText: "Checks if a user is an admin.",
 		Description: `
 		Checks if a user is an admin.
 		`,
-		After: telemetry.AfterRunAction(),
-		Flags: []cli.Flag{
-			&flags.VerboseFlag,
-			&AccountAddressFlag,
-			&CallerAddressFlag,
+		Action: func(c *cli.Context) error {
+			return isAdmin(c, readerGenerator)
 		},
+		After: telemetry.AfterRunAction(),
+		Flags: IsAdminFlags(),
+	}
+
+	return cmd
+}
+
+func isAdmin(cliCtx *cli.Context, generator func(logging.Logger, *isAdminConfig) (IsAdminReader, error)) error {
+	ctx := cliCtx.Context
+	logger := common.GetLogger(cliCtx)
+
+	config, err := readAndValidateIsAdminConfig(cliCtx, logger)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to read and validate user can call config", err)
+	}
+	cliCtx.App.Metadata["network"] = config.ChainID.String()
+	elReader, err := generator(logger, config)
+	if err != nil {
+		return err
+	}
+
+	result, err := elReader.IsAdmin(ctx, config.AccountAddress, config.AdminAddress)
+	if err != nil {
+		return err
 	}
+	fmt.Printf("IsAdmin Result: %v\n", result)
```

Comment:
maybe can be a better message like "<adminaddress> is an admin on account <accountaddress>" or something like that